Missionframework/modules/15_build/ui/buildDisplay_left.hpp

```diff
@@ -0,0 +1,96 @@
+
+class KPLIB_BuildCategoryList: KPGUI_PRE_Combo {
+    idc = KPLIB_IDC_BUILD_CATEGORY_LIST;
+
+    x = KP_GETCX(KP_X_VAL_LP,KP_WIDTH_VAL_LP,0,1);
+    y = KP_GETCY(KP_Y_VAL_LP,KP_HEIGHT_VAL_LP,2,40);
+    w = KP_GETW(KP_WIDTH_VAL_LP,1);
+    h = KP_GETH(KP_HEIGHT_VAL_LP,40);
+
+    onMouseZChanged = "['onMouseZChanged_BuildCategoryList', _this] call KPLIB_fnc_build_handleMouse";
+};
+
+class KPLIB_BuildSearch: KPGUI_PRE_EditBox {
+    idc = KPLIB_IDC_BUILD_SEARCH;
+
+    x = KP_GETCX(KP_X_VAL_LP,KP_WIDTH_VAL_LP,0,12);
+    y = KP_GETCY(KP_Y_VAL_LP,KP_HEIGHT_VAL_LP,3,40);
+    w = KP_GETW(KP_WIDTH_VAL_LP,(12/11));
+    h = KP_GETH(KP_HEIGHT_VAL_LP,40);
+
+    onKeyUp = "call KPLIB_fnc_build_displayFillList";
+};
+
+class KPLIB_BuildSearchClearButton: KPGUI_PRE_ActivePictureRatio {
+    idc = KPLIB_IDC_BUILD_SEARCH_BUTTON;
+
+    x = KP_GETCX(KP_X_VAL_LP,KP_WIDTH_VAL_LP,11,12);
+    y = KP_GETCY(KP_Y_VAL_LP,KP_HEIGHT_VAL_LP,3,40);
+    w = KP_GETW(KP_WIDTH_VAL_LP,(12/1));
+    h = KP_GETH(KP_HEIGHT_VAL_LP,40);
+
+    text = "\a3\Ui_f\data\GUI\RscCommon\RscButtonSearch\search_end_ca.paa";
+    action = "call KPLIB_fnc_build_searchClear";
```

Comment:
```suggestion

```
Can be removed as the onKeyUp event doesn't block the backspace/delete keyboard button.

Reply:
I left it as it's convivient to have clear search button IMO.

Reply:
I left it as it's convivient to have clear search button IMO.